### PR TITLE
transform/base64: check for 0-sized buffer

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/7296

Describe changes:
- transform/base64: check for 0-sized buffer

As done already by other transforms

### CODE REVIEW

File: src/detect-transform-base64.c [hunk 1]
```diff
@@ -120,6 +120,10 @@ static void TransformFromBase64Decode(InspectionBuffer *buffer, void *options)
     const uint32_t input_len = buffer->inspect_len;
     uint32_t decode_length = input_len;
 
+    if (input_len == 0) {
+        return;
+    }
```
Comment (inashivb) [inline src/detect-transform-base64.c:125]: It looks better indeed but where do we create the variable length array?

**Comment by victorjulien** [comment]
Is there a meaningful SV test we can do? I guess its at least good to have one that passes with the UBSAN flag enabled.

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .uptime | 649 | 629 | 96.92% | 

Pipeline 22966

File: src/detect-transform-base64.c (hunk 1 above)
Comment (catenacyber) [reply src/detect-transform-base64.c:125]: `uint8_t decoded[decoded_size];` below

**Comment by catenacyber** [comment]
> Is there a meaningful SV test we can do? I guess its at least good to have one that passes with the UBSAN flag enabled.

Done in https://github.com/OISF/suricata-verify/pull/2075 and new version in https://github.com/OISF/suricata/pull/11869

**Pull request closed without merging**